server/src/test/java/org/elasticsearch/test/BackgroundIndexer.java

```diff
@@ -0,0 +1,321 @@
+     * been indexed.
+     *
+     * @param table       table name to write data
+     * @param column      column name to write data
+     * @param numOfDocs   number of document to index before pausing. Set to -1 to have no limit.
+     * @param writerCount number of indexing threads to use
+     * @param autoStart   set to true to start indexing as soon as all threads have been created.
+     * @param random      random instance to use
+     */
+    public BackgroundIndexer(final String table, final String column, final SQLTransportExecutor sqlExecutor, final int numOfDocs, final int writerCount,
+                             boolean autoStart, Random random) {
+        if (random == null) {
+            random = RandomizedTest.getRandom();
+        }
+        this.sqlExecutor = sqlExecutor;
+        writers = new Thread[writerCount];
+        stopLatch = new CountDownLatch(writers.length);
+        logger.info("--> creating {} indexing threads (auto start: [{}], numOfDocs: [{}])", writerCount, autoStart, numOfDocs);
+        for (int i = 0; i < writers.length; i++) {
+            final int indexerId = i;
+            final boolean batch = random.nextBoolean();
+            final Random threadRandom = new Random(random.nextLong());
+            writers[i] = new Thread() {
+                @Override
+                public void run() {
+                    long id = -1;
+                    try {
+                        startLatch.await();
+                        logger.info("**** starting indexing thread {}", indexerId);
+                        while (!stop.get()) {
+                            if (batch) {
+                                int batchSize = threadRandom.nextInt(20) + 1;
+                                if (hasBudget.get()) {
+                                    // always try to get at least one
+                                    batchSize = Math.max(Math.min(batchSize, availableBudget.availablePermits()), 1);
+                                    if (!availableBudget.tryAcquire(batchSize, 250, TimeUnit.MILLISECONDS)) {
+                                        // time out -> check if we have to stop.
+                                        continue;
+                                    }
+                                }
+                                var insertData = new Object[batchSize];
+                                for (int i = 0; i < batchSize; i++) {
+                                    insertData[i] = generateValues(idGenerator.incrementAndGet(), threadRandom);
+                                }
+                                try {
+                                    var insert = String.format(Locale.ENGLISH,"insert into %s (id, %s) (select * from unnest(?)) returning _id", table, column);
+                                    var response = sqlExecutor.execute(insert, insertData).actionGet();
```

Comment:
```suggestion
                                    var response = sqlExecutor.exec(insert, insertData);
```
Otherwise, please also keep in mind to always add proper timeouts to any blocking calls in general to let if fail early.